### PR TITLE
Project-wide Checkstyle Config

### PR DESCRIPTION
**Pull request by mgodave**
### Motivation

Checkstyle allows us to enforce project norms and safe coding practices. 

### Modifications

Add project-wide checkstyle config to root pom.xml, add buildtools project to house config. The config is taken from bookkeeper as a starting point.

### CODE REVIEW

File: buildtools/pom.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  <parent>
+    <groupId>org.apache.pulsar</groupId>
+    <artifactId>pulsar</artifactId>
+    <version>1.22.0-incubating-SNAPSHOT</version>
+    <relativePath>..</relativePath>
+  </parent>
+
+  <artifactId>buildtools</artifactId>
+  <packaging>jar</packaging>
+  <name>Pulsar Checkstyle Config</name>
```
Comment (sijie) [inline buildtools/pom.xml]: nit: maybe "Pulsar Build Tools"  is better?

File: managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/EntryCacheTest.java [hunk 2]
```diff
@@ -20,19 +20,14 @@
 
 import static org.mockito.Matchers.any;
 import static org.mockito.Matchers.anyLong;
-import static org.mockito.Mockito.doAnswer;
-import static org.mockito.Mockito.doReturn;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.never;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
```
Comment (sijie) [inline managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/EntryCacheTest.java]: might be good to make this consistent, either use "import *" or just disallow importing "*". allow mixing them together is not very good.

Comment (mgodave) [reply managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/EntryCacheTest.java]: This change was actually a mistake. The Checkstyle config explicitly disallows star imports. I'll remove it.

**Comment by mgodave** [comment]
OOF, I messed this up, I had meant to just add the config and the managed-ledger changes in a separate PR. Let me fix this.

**Comment by mgodave** [comment]
@sijie fixed the comment on naming and restricted the changes to just adding the checkstyle config and infrastructure. Once this is merged I will submit the PR for the managed-ledger changes.

**Comment by merlimat** [comment]
@mgodave Change LGTM. Will this start enforcing the style on new PRs or will need to be enabled module by module after fixing them?

**Comment by mgodave** [comment]
Enabled module by module as they are fixed.

**Pull request merged**